tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -2474,7 +2517,12 @@ private object FunctionEmitter {
   ) {
     def isLocalVar(ident: Ident): Boolean = vars.contains(ident.name)
 
-    def isLocalMutable(ident: Ident): Boolean = vars(ident.name)
+    def isLocalMutable(ident: Ident): Boolean = {
+      vars.getOrElse(ident.name, {
+        // It must be a JS class capture, which must be immutable
+        false
+      })
```

Comment:
This syntax is quite odd.

Have you considered:
```scala
// If we do not know the var, it must be a JS class capture, which must be immutable
vars.getOrElse(ident.name, false)
```

Reply:
Doesn't seem odd to me given that `getOrElse` takes a by-name argument for `default`. But I can change it.